### PR TITLE
Created doc: EXPLAIN and PROFILE

### PR DESCRIPTION
**Pull request by randomJoe211**

### CODE REVIEW

**Review by whitewum — APPROVED** [review]

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+The execution plan is determined by the nGQL execution planner. nGQL execution planner parses nGQL statements into subtasks and assigns them to several types of plan nodes. 
+
+For example, a simple `SHOW TAGS` statement is parsed as two subtasks and assigned to a `Start` plan node and a `ShowTags` plan node, while a more complex `GO` statement may be parsed as more than 10 subtasks and assigned to plan nodes of the same number.
+
+## Syntax
+
+* `EXPLAIN`
+
+```nGQL
+EXPLAIN [format="dot"] <your nGQL statement>
```
Comment (yixinglu) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:16]: `format="dot|row"`, default value is `"row"`

**Review by yixinglu — COMMENTED** [review]
Well Done

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md (hunk 1 above)
Comment (randomJoe211) [reply docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md]: Thank you and updated.

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+|-|-|
+|id|Indicates the ID of the operator.|
+|name|Indicates the name of the operator.|
+|dependencies|Shows the ID of the operator that the current operator depends on.|
+|profiling data|Shows the execution profile. `ver` is the version of the operator, which you can use to identify loops; `rows` shows the number of rows to be output by the operator; `execTime` shows the execution time only; `totalTime` contains not only the execution time, but also the time cost by system scheduling and queueing.|
+|operator info|Shows the detailed information of the operator.|
+
+## Format "dot"
+
+You can use the `format="dot"` option to output the return message in the DOT language, then use Graphviz to generate a graph of the plan.
```
Comment (oldLady344) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:88]: , and then

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+# EXPLAIN and PROFILE
+
+`EXPLAIN` helps output the execution plan of an nGQL statement without executing the statement, while `PROFILE` executes the statement, then output the execution plan as well as the execution profile. You can optimize the queries for better performance with the execution plan and profile.
```
Comment (oldLady344) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:3]: splitting the first sentence into two sentences is suggested, which could exert less stress to readers.

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+## Execution Plan
+
+The execution plan is determined by the execution planner in the Nebula Graph query engine.
+
+Execution planner processes the parsed nGQL statements into actions. An action is the smallest unit that can be executed. A typical action could be fetching all neighbors of a given vertex, getting the properties of an edge, or filtering vertices or edges based on the given condition. Each action is assigned to a operator which performs the action.
```
Comment (oldLady344) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:9]: assigned to a operator==>assigned to an operator

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+* `PROFILE`
+
+```nGQL
+PROFILE [format="row" | "dot"] <your_nGQL_statement>
+```
+
+## Output formats
+
+The return message of an `EXPLAIN` or a `PROFILE` statement has two formats, the default format "row" and format "dot". You can use the `format` option to modify the output format.
```
Comment (oldLady344) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:29]: the default format "row" and format "dot" ==> the default "row" format and the "dot" format?

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  +----+----------+--------------+----------------------------------------------------+---------------------------------------------------------------------+
+  |  2 | Start    |              | ver: 0, rows: 0, execTime: 1us, totalTime: 57us    | outputVar: [{"colNames":[],"name":"__Start_2","type":"DATASET"}]    |
+  +----+----------+--------------+----------------------------------------------------+---------------------------------------------------------------------+
+  ```
+
+The descriptions of the columns are as follows:
+
+|Column|Description|
+|-|-|
+|id|Indicates the ID of the operator.|
```
[the remark this answers is not in the document]
Comment (randomJoe211) [reply docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:80]: of the operator [in that row]. So it should be "the"

**Review by whitewum — APPROVED** [review]

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+The return message of an `EXPLAIN` or a `PROFILE` statement has two formats, the default "row" format and the "dot" format. You can use the `format` option to modify the output format.
+
+## Format "row"
+
+The "row" format outputs the return message in a table as follows.
+
+* `EXPLAIN`:
+
+  ```nGQL
+  (root@nebula) [nba]> EXPLAIN SHOW TAGS
```
Comment (Amber1990Zhang) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:38]: EXPLAIN [format="row"]  SHOW TAGS

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+The return message of an `EXPLAIN` or a `PROFILE` statement has two formats, the default "row" format and the "dot" format. You can use the `format` option to modify the output format.
+
+## Format "row"
+
+The "row" format outputs the return message in a table as follows.
+
+* `EXPLAIN`:
+
+  ```nGQL
+  (root@nebula) [nba]> EXPLAIN [format="row"] SHOW TAGS
```
Comment (whitewum) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:38]: []

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
+# EXPLAIN and PROFILE
+
+`EXPLAIN` helps output the execution plan of an nGQL statement without executing the statement. `PROFILE` executes the statement, then output the execution plan as well as the execution profile. You can optimize the queries for better performance with the execution plan and profile.
+
+## Execution Plan
+
+The execution plan is determined by the execution planner in the Nebula Graph query engine.
+
+Execution planner processes the parsed nGQL statements into actions. An action is the smallest unit that can be executed. A typical action could be fetching all neighbors of a given vertex, getting the properties of an edge, or filtering vertices or edges based on the given condition. Each action is assigned to an operator which performs the action.
```
Comment (oldLady344) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:9]: the given condition.==> "the given conditions."  OR "a given condition."

Comment (oldLady344) [reply docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:9]: which performs the action. ==> that performs the action.

THAT is preferred to introduce an essential adjective clause.

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+For example, a `SHOW TAGS` statement is processed as two actions and assigned to a `Start` operator and a `ShowTags` operator, while a more complex `GO` statement may be processed as more than 10 actions and assigned to operators of the same number.
```
Comment (oldLady344) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:11]: as two actions and assigned to a `Start` operator and a `ShowTags` operator, ==> into(?) two actions that are assigned to a `Start` operator and a `ShowTags` operator

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md (hunk 9 above)
Comment (randomJoe211) [reply docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:9]: Thanks and fixed.

**Review by yixinglu — APPROVED** [review]

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
+# EXPLAIN and PROFILE
+
+`EXPLAIN` helps output the execution plan of an nGQL statement without executing the statement. `PROFILE` executes the statement, then output the execution plan as well as the execution profile. You can optimize the queries for better performance with the execution plan and profile.
```
Comment (oldLady344) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:3]: executes the statement, then output the ==>executes a statement, then outputs the

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+
+## Execution Plan
+
+The execution plan is determined by the execution planner in the Nebula Graph query engine.
+
+Execution planner processes the parsed nGQL statements into actions. An action is the smallest unit that can be executed. A typical action could be fetching all neighbors of a given vertex, getting the properties of an edge, or filtering vertices or edges based on the given conditions. Each action is assigned to an operator that performs the action.
```
Comment (oldLady344) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:9]: Execution planner processes ==> The execution planner processes ?

File: docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+  ```
+
+The descriptions of the columns are as follows:
+
+|Column|Description|
+|-|-|
+|`id`|Indicates the ID of the operator.|
+|`name`|Indicates the name of the operator.|
+|`dependencies`|Shows the ID of the operator that the current operator depends on.|
+|`profiling data`|Shows the execution profile. `ver` is the version of the operator, which you can use to identify loops; `rows` shows the number of rows to be output by the operator; `execTime` shows the execution time only; `totalTime` contains not only the execution time, but also the time cost by system scheduling and queueing.|
```
Comment (oldLady344) [inline docs_2.0/manual_EN/4.ngql-guide/16.query-tuning-statements/1.explain-and-profile.md:83]: the execution time, but also the time ==>the execution time but also the time

**Review by whitewum — APPROVED** [review]

**Pull request merged**